aws/templates/resource/resource_waf.ftl

```diff
@@ -69,13 +236,37 @@
     /]
 [/#macro]
 
-[#macro createWAFAcl mode id name metric default rules]
+[#-- Rules should be ordered in decreasing order of importance --]
+[#macro createWAFAcl mode id name metric default rules=[] regional=false]
```

Comment:
Should we have priorities here?

Reply:
Priorities set to reflect rules array ordering.  Each to add a Priority attribute to the rule object, but then all of the rules would need one to avoid ambiguity.

Reply:
Happy to add if you have a use case in mind. Was waiting for one before adding the extra complexity. Might find it is important if mixing predefined rules with extras. We'd need to control priority ranges then.

Reply:
Actually scratch that since use of rule is when priority added.

Reply:
Only a behaviour thing really. If i was adding a new rule my general thinking would be to add it to the end. which would make it the last in the list 

Since the priority here isn't going through to AWS just ordering the list based on priority would allow for overlaps but give you reasonable control over the rule order. 

ec2 init scripts have a similar function at the moment 
https://github.com/codeontap/gen3/blob/master/aws/templates/resource/resource_ec2.ftl#L39 

the idea here was that I can set priorities in groups to say, this rule must be processed befoer another one, but I'm not fussed of the order with scripts in the same prirotiy

Reply:
So your idea is more of a "band" thn a specific priority